src/impl/collection_notifier.hpp

```diff
@@ -49,6 +49,16 @@ struct TransactionChangeInfo {
     std::vector<ListChangeInfo> lists;
     std::vector<CollectionChangeBuilder> tables;
     bool track_all = false;
+
+    // Required by GCC 4.9
+    TransactionChangeInfo() {}
+    TransactionChangeInfo(std::vector<bool> table_mods,
+                          std::vector<bool> table_moves,
+                          std::vector<ListChangeInfo> ls)
+            : table_modifications_needed(table_mods),
```

Comment:
These all should be moved rather than copied (i.e. `table_modifications_needed(std::move(table_mods))` etc.).